app/src/main/java/fr/free/nrw/commons/nearby/NearbyListFragment.java

```diff
@@ -32,80 +26,103 @@
 
 import fr.free.nrw.commons.R;
 
-public class NearbyListFragment extends ListFragment {
+public class NearbyListFragment extends ListFragment implements TaskListener {
 
-    private int mImageSize;
-    private boolean mItemClicked;
+    private NearbyAsyncTask nearbyAsyncTask;
     private NearbyAdapter mAdapter;
+    private ListView listview;
+
+    private ProgressBar progressBar;
```

Comment:
Are you planning to create a progress bar?
Would be useful indeed, even though probably low-priority for now as CSV download might be discarded in favour of another API.